README.md

```diff
@@ -0,0 +1,143 @@
+# Collections for PHP with focus on Immutability and Functional Programming
+[![Build Status](https://github.com/bonami/collections/workflows/CI/badge.svg)
+[![Latest Stable Version](https://poser.pugx.org/bonami/collections/v/stable)](https://packagist.org/packages/bonami/collections)
+[![License](https://poser.pugx.org/bonami/collections/license)](https://packagist.org/packages/bonami/collections)
+
+**TODO** table of contents
+
+## Motivation
+
+Why yet another collections library for PHP? Native PHP arrays or SPL structures like SplFixedArray or SplObjectStorage(and other) are mutable and has very strange interfaces and behaviors. They often represents more data structures at once (eg. SplObjectStorage represents both Set and Map) and theirs interfaces are designed for classic imperative approach.
+
+We tried to design interfaces of our structures, to be focused on declarative approach leveraging functional programing. For more safety, we designed structures to be immutable (we have some mutables as well, because sometime it is necessary for optimization reasons)
```

Comment:
`structures to be focused ` (tam asi carka nema byt)

Reply:
`we have some mutable variants as well` .. `because sometimes `

A mozna bych pouzil misto `optimization reasons` -> `performance reasons` ?

Reply:
comma removed, performance wording used.